openmc/deplete/flux_operator.py

```diff
@@ -0,0 +1,594 @@
+
+            self.nuc_ind_map = {ind: nuc for nuc, ind in rates.index_nuc.items()}
+            self.rxn_ind_map = {ind: rxn for rxn, ind in rates.index_rx.items()}
+            self._op = op
+
+        def generate_tallies(self, materials, scores):
+            """Unused in this case"""
+            pass
+
+        def get_material_rates(self, mat_id, nuc_index, react_index):
+            """Return 2D array of [nuclide, reaction] reaction rates
+
+            Parameters
+            ----------
+            mat_id : int
+                Unique ID for the requested material
+            nuc_index : list of str
+                Ordering of desired nuclides
+            react_index : list of str
+                Ordering of reactions
+            """
+            self._results_cache.fill(0.0)
+
+            volume = self._op.number.get_mat_volume(mat_id)
+            for i_nuc, i_react in product(nuc_index, react_index):
+                nuc = self.nuc_ind_map[i_nuc]
+                rxn = self.rxn_ind_map[i_react]
+                density = self._op.number.get_atom_density(mat_id, nuc)
+                self._results_cache[i_nuc,
+                                    i_react] = self._op.cross_sections[rxn][nuc] * density * volume
+
+            return self._results_cache
+
+    def _get_helper_classes(self, helper_kwargs):
+        """Get helper classes for calculating reation rates and fission yields
+
+        Parameters
+        ----------
+        helper_kwargs : dict
+            Keyword arguments for helper classes
+
+        """
+
+        normalization_mode = helper_kwargs['normalization_mode']
+        fission_yield_opts = helper_kwargs.get('fission_yield_opts', {})
+
+        self._rate_helper = self._FluxDepletionRateHelper(
+            self.reaction_rates.n_nuc, self.reaction_rates.n_react, self)
+        if normalization_mode == "constant-power":
+            self._normalization_helper = self._FluxDepletionNormalizationHelper(self)
+        else:
+            self._normalization_helper = SourceRateHelper()
+
+        # Select and create fission yield helper
+        fission_helper = ConstantFissionYieldHelper
+        if fission_yield_opts is None:
+            fission_yield_opts = {}
+        self._yield_helper = fission_helper.from_operator(
+            self, **fission_yield_opts)
+
+    def initial_condition(self):
+        """Performs final setup and returns initial condition.
+
+        Returns
+        -------
+        list of numpy.ndarray
+            Total density for initial conditions.
+        """
+
+        # Return number density vector
+        return super().initial_condition(self.materials)
+
+    def __call__(self, vec, source_rate):
+        """Obtain the reaction rates
+
+        Parameters
+        ----------
+        vec : list of numpy.ndarray
+            Total atoms to be used in function.
+        source_rate : float
+            Power in [W] or source rate in [neutron/sec]
```

Comment:
I've replaced `source rate` with flux and changed the units, because passing a source rate to this class isn't quite correct.

Should I add a warning statement to the `Integrator` class to alert a user that they may get unexpected results if passing a `source_rate` parameter instead of `flux` or `power` if the `operator` is a `FluxDepletionOperator` object?